src/bundles/ipfs.js

```diff
@@ -27,5 +38,21 @@ export default {
     }
 
     dispatch({ type: 'IPFS_INIT_FINISHED' })
+  },
+
+  doGetGatewayUrl: () => async ({ dispatch }) => {
```

Comment:
`doGetGatewayUrl` isn't called? I think it's a good plan to have something like this. We might end up with both a "publicGatewayUrl" that points to https://ipfs.io most of the time, and an "localGatewayUrl" that usually points to localhost. I think we've got something like that in companion already. 

What you have is good for now. We need to think about how we make this work against a js-ipfs instance, as we connect to window.ipfs, so we can't be sure that the user has http gateway. We could just use the public one in that case, but it would be neat if we could load the users files directly via window.ipfs rather than assuming an http gateway.

Reply:
Uhh, forgot to call it. Now it's called after doInitIPFS. 😄 It doesn't work when using companion though because it doesn't allow `.config.get`.

Reply:
We should start thinking on adding `window.ipfs` in "privileged mode" for WebUI, which provides access to all APIs and has no sandboxing. 

Initial idea was to add a mechanism for pre-emptively requesting permissions, as suggested in https://github.com/ipfs-shipyard/ipfs-companion/issues/454#issuecomment-381245084 and https://github.com/ipfs-shipyard/ipfs-companion/issues/454#issuecomment-381278859, it could be extended with sandbox flag:

    window.ipfs.proxy.acl.whitelist(['config.get'], {sandboxing: false})

The problem is that other dapps could easily disable sandboxing this way. 
Not sure if we are able to secure it without some sort of pubkey crypto, which requires us to maintain key for signing webui reuest. 

Is this a blocker or just an inconvenience? 
Are you able to work with HTTP API until we sort this out?